File: iOS/Objective-C/SampleObjC/AuthenticationHelper.m
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.
#import "AuthenticationHelper.h"

// You can use this helper method to get an authentication token via Azure Active Directory.
// For getting going quickly, you can instead set the SpatialAnchorsAccountId and SpatialAnchorsAccountKey in ViewController.m.

// Set these strings to the Service URL provided for the Azure Spatial Service resource.
NSString *const AuthServiceBaseUrl = @"https://sts.mixedreality.azure.com";
NSString *const ClientId = @"Set me";
NSString *const TenantId = @"Set me";

@implementation AuthenticationHelper {
}

+ (void) acquireAuthenticationToken:(void (^)(NSString *token, NSError *error))completionBlock
{
    ADAuthenticationError *error;
    ADAuthenticationContext *authContext = [ADAuthenticationContext authenticationContextWithAuthority:[NSString stringWithFormat:@"https://login.microsoftonline.com/%@", TenantId] error:&error];
    NSURL *redirectUri = [[NSURL alloc]initWithString:@"urn:ietf:wg:oauth:2.0:oob"];
    
    [authContext acquireTokenWithResource:AuthServiceBaseUrl
                                 clientId:ClientId
                              redirectUri:redirectUri
                          completionBlock:^(ADAuthenticationResult *result) {
                              
                              if (result.status != AD_SUCCEEDED)
                              {
                                  completionBlock(nil, result.error);
                              }
                              else
                              {
                                  completionBlock(result.accessToken, nil);
                              }
                          }];
}

@end

File: iOS/Swift/SampleSwift/AuthenticationHelper.swift
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.

import ADAL
import Foundation

// You can use this helper method to get an authentication token via Azure Active Directory.
// For getting going quickly, you can instead set the spatialAnchorsAccountId and spatialAnchorsAccountKey in BaseViewController.swift
class AuthenticationHelper {
    
    // Set these strings to the Service URL provided for the Azure Spatial Service resource.
    static let AuthServiceBaseUrl = "https://sts.mixedreality.azure.com"
    static let ClientId = "Set me"
    static let TenantId = "Set me"
    
    static func acquireAuthenticationToken(completion: @escaping (_ accessToken: String?, Error?) -> Void) {
        let authContext = ADAuthenticationContext(authority: "https://login.microsoftonline.com/" + TenantId,
                                                  error: nil)
        
        authContext!.acquireToken(withResource: AuthServiceBaseUrl,
                                  clientId: ClientId,
                                  redirectUri: URL(string: "urn:ietf:wg:oauth:2.0:oob")) {
            (result) in
            
            if (result!.status != AD_SUCCEEDED) {
                completion(nil, result!.error)
            }
            else {
                completion(result!.accessToken, nil)
            }
        }
    }
}
